tests/doc/test-levels.md

```diff
@@ -0,0 +1,150 @@
+# Test scopes
+
+## Overview
+
+This document is an analysis of testing scopes, current and proposed, in the SDK as they
+relate to [EPIC: Separate all SDK modules into standalone go modules](https://github.com/cosmos/cosmos-sdk/issues/11899).  Some tests have
+been written such that they introduce additional dependencies for the test phase only, and
+these must be addressed for the successful completion of Phases 3 and 4.
+
+## Unit tests
+
+Unit tests exercise the code contained in a single module (e.g. `/x/bank`) or package
+(e.g. `/client`) in isolation from the rest of the code base.  If an SDK component has
+dependencies on other modules they must be mocked.
+
+Example 1 [Bank keeper tests](https://github.com/cosmos/cosmos-sdk/blob/2bec9d2021918650d3938c3ab242f84289daef80/x/bank/keeper/keeper_test.go#L94-L105) - A mock implementation of `AccountKeeper` is
+supplied to the keeper constructor.
+
+Example 2 [depinject tests](https://github.com/cosmos/cosmos-sdk/blob/main/depinject/provider_desc_test.go)
+
+#### Limitations
+
+Certain modules are tightly coupled beyond the test phase.  See the dependency report
+for [bank->auth](./test-levels/bank-auth.txt), where 274 total usages of `x/auth` were found in `x/bank`, 50 of
+which are in production code and 224 in test.  This tight coupling may suggest that
+either the modules should be merged, or refactoring is required to abstract references
+to the core types tying the modules together.  It could also indicate that these
+modules should be tested together in integration tests beyond mocked unit tests.
+
+Setting up a test case for a module with many mocked dependencies is some cases is quite cumbersome
+and the resulting test may only show that the mocking framework works as expected rather than
+working as a functional test of interdependent module behavior.
+
+#### Proposal
+
+Continue to rewrite existing integration tests into mocked unit tests as per [EPIC: Unit testing of
+modules via mocks](https://github.com/cosmos/cosmos-sdk/issues/12398) but maintain current integration tests, moving them to `/tests/integration`.
```

Comment:
I'd like to see a small note about code duplication if we go with this proposal: exact same tests will exist twice: in the module (with mocks) and in `tests/integration`.

I think the way to address this is on a per-module basis: it's probably fine to keep duplicated tests in some modules (bank, evidence), but it's also worth exploring rewriting some modules' unit tests from scratch. BDD was mentioned as a new way.